shared/js/sk_key_utils.js

```diff
@@ -0,0 +1,367 @@
+(function (global) {
+  'use strict';
+
+  var softKeyModule = (function () {
+
+    // Keys which are mapped to accesskey
+    const SK_KEYS_MAPPING = {
+      LSK: 'a',
+      CSK: 's',
+      RSK: 'd'
+    };
+
+    // Storing the reference to softkeys
+    var sk_keys = {
+      LSK: null,
+      CSK: null,
+      RSK: null
+    };
+
+    // Container created for SoftKey
+    var sk_container = null;
+
+    // Check if menu open
+    var sk_options_menu = {
+      open: false,
+      cancelKey: SK_KEYS_MAPPING.LSK,
+      cancelKeyText: 'Cancel',
+      cancelKeyl10nId: 'cancel'
+    };
+
+    // Observer for observing the DOM changes
+    var observer = null;
+
+    return {
+      init: init
+    };
+
+    /**
+     * Used to initialize the library
+     * @example
+     * softKeyModule.init();
+     *
+     * @param N/A
+     * @return void 0
+     */
+    function init() {
+
+      // Calling destroy to destroy previously initialized variables
+      destroy();
+
+      // Selecting the softkey DOM element
+      var softKey = document.querySelector('[role="softkey"]');
+      if (!softKey) {
+        // If null, return
+        return;
+      }
+
+      // Container will be created to handle options
+      // menu and soft keys
+      createSoftKeyContainer();
+
+      for (var k in sk_keys) {
+        if (sk_keys.hasOwnProperty(k)) {
+
+          // Selecting the button using accesskey
+          sk_keys[k] =
+                  softKey.querySelector('[accesskey="' + SK_KEYS_MAPPING[k] + '"]');
```

Comment:
Most of our Gaia apps are single-page apps where multiple views are defined in the document, but only one is visible at a time.  So a full implementation of this soft key system may need to use querySelectorAll() to find all elements with this accesskey and then check to see which one is visible  (or which one is in a container that is visible).